### PR TITLE
SISRP-17144 Do not create unusable link to Schedule Planner

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.berkeley.edu/browse/SISRP-17144

### CODE REVIEW

File: app/controllers/campus_solutions/campus_solutions_controller.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def delete_passthrough(classname)
     def exclude_acting_as_users
       return unless Settings.features.prevent_acting_as_users_from_posting
       unless current_user.directly_authenticated?
-        logger.warn "ACT-AS: User #{current_user.original_user_id} attempted access to an endpoint which is forbidden while acting-as user #{current_user.user_id}"
+        logger.warn "ACT-AS: User #{current_user.real_user_id} attempted access to an endpoint which is forbidden while acting-as user #{current_user.user_id}"
```
Comment (raydavis) [inline app/controllers/campus_solutions/campus_solutions_controller.rb]: The old code would not have shown Advisor or Delegate UIDs.

**Comment by christianvuerings** [comment]
Change lgtm - I'll also let @paulkerschen  or @johncrossman have a look at this one.

File: src/assets/templates/widgets/enrollment/schedule.html [hunk 2]
```diff
@@ -16,7 +16,7 @@
   </div>
 
   <div
-    data-ng-if="!api.user.profile.actAsOptions.canSeeCSLinks"
+    data-ng-if="!api.user.profile.actAsOptions.isDirectlyAuthenticated"
     data-ng-include="'widgets/enrollment/enrollment_hide_links.html'">
```
Comment (johncrossman) [inline src/assets/templates/widgets/enrollment/schedule.html]: Do we need same treatment for decide.html, explore.html and plan.html in directory `src/assets/templates/widgets/enrollment`?

Comment (johncrossman) [inline src/assets/templates/widgets/enrollment/schedule.html]: After a closer look, I think the answer is no. I'll let @raydavis confirm.

Comment (raydavis) [inline src/assets/templates/widgets/enrollment/schedule.html]: I've asked for the folks what should know to tell me what other magic URLs need more protection. I haven't heard of any yet and I don't think we should wait to resolve the blocker. In other words, confirmed. :)

**Comment by johncrossman** [comment]
Thanks, @raydavis  The known knowns are verified. Unknown unknowns are still out there.

**Pull request merged**